src/importlinter/domain/helpers.py

```diff
@@ -57,32 +57,38 @@ def import_expression_to_imports(
         MissingImport if an import is not present in the graph. For a wildcarded import expression,
         this is raised if there is not at least one match.
     """
-    imports: Set[DirectImport] = set()
-    matched = False
+    matching_imports = graph.find_matching_direct_imports(import_expression=str(expression))
 
-    importers = module_expression_to_modules(graph, expression.importer)
-    importeds = module_expression_to_modules(graph, expression.imported)
-    for importer, imported in itertools.product(importers, importeds):
-        import_details = graph.get_import_details(importer=importer.name, imported=imported.name)
+    if not matching_imports:
+        raise MissingImport(
+            f"Ignored import expression {expression} didn't match anything in the graph."
+        )
+
+    detailed_imports: Set[DirectImport] = set()
+    for matching_import in matching_imports:
+        import_details = graph.get_import_details(
+            importer=matching_import["importer"],
+            imported=matching_import["imported"],
+        )
 
         if import_details:
             for individual_import_details in import_details:
-                imports.add(
+                detailed_imports.add(
                     DirectImport(
                         importer=Module(individual_import_details["importer"]),
                         imported=Module(individual_import_details["imported"]),
                         line_number=individual_import_details["line_number"],
                         line_contents=individual_import_details["line_contents"],
                     )
                 )
-            matched = True
-
-    if not matched:
-        raise MissingImport(
-            f"Ignored import expression {expression} didn't match anything in the graph."
-        )
-
-    return list(imports)
+        else:
+            # TODO(peter)
+            # This case was previously unhandled. How to handle this case?
+            # Add DirectImport with line_number/line_contents of `None`?
+            # It's possible for imports to exist in the graph without details.
+            pass
```

Comment:
The behaviour here is unchanged. Before we were simply skipping over imports without details. I feel like we probably should do something here. Options I see:

* Raise an error (consistent with `pop_imports`, see https://github.com/seddonym/import-linter/pull/257#discussion_r1986723093)
* Or, return a `import_linter.DirectImport(...)` with `line_number` and `line_contents` of `None`.

Should I change anything here, for this PR?

Reply:
> Before we were simply skipping over imports without details.

I don't think there will be any imports without details, so I think it's okay. Shall we remove the `else` block?